nemo/collections/nlp/modules/common/megatron/megatron_bert.py

```diff
@@ -78,13 +83,28 @@ def __init__(self, model_name, config, vocab_file, model_parallel_size=None, mod
             os.environ["RANK"] = str(self._model_parallel_rank)
 
             # used to set model_parallel_size in megatron-lm argparser
-            def _update_model_parallel_arg(parser):
+            def _update_megatron_args(parser):
                 parser.set_defaults(model_parallel_size=self._model_parallel_size)
-                return parser
+                parser.set_defaults(micro_batch_size=1)
+                parser.set_defaults(scaled_masked_softmax_fusion=False)
+                parser.set_defaults(bias_gelu_fusion=False)
+                parser.set_defaults(bias_dropout_fusion=False)
 
-            extra_args_provider = _update_model_parallel_arg
         else:
-            extra_args_provider = None
+            # megatron expects batch size is not None
+            def _update_megatron_args(parser):
+                parser.set_defaults(micro_batch_size=1)
+                parser.set_defaults(scaled_masked_softmax_fusion=False)
+                parser.set_defaults(bias_gelu_fusion=False)
+                parser.set_defaults(bias_dropout_fusion=False)
+
+                return parser
```

Comment:
Could you please combine two _update_megatron_args() methods and add an extra param `model_parallel_size`

Reply:
Done.